how-to: Node-RED filter node (#862)
flowforge/website: src/blog/2023/06/node-explained-filter.md

---
title: "Nodes explained: Filter"
subtitle: The filter node is incredibly versatile, let's go over some use-cases
description: Forwarding or blocking messages based on values, change of them, or lack of change.

> date: 2023-06-15

Author: ```suggestion
date: 2023-06-16
```

authors: ["zeger-jan-van-de-weg"]

> image:

Author: Not sure if this is worth the time for an art request? Or if a generated image is fine?

Reviewer: I agree it is not worth it.

tags:
  - posts
  - how-to
---

The Filter node, previously called "Report by Exception", has two modes of
operation, called deadband and narrowband. These modes allow users to limit network

> traffic or write operations to historians. Or limit reporting of values outside

Reviewer: ```suggestion
traffic, write operations to historians, or limit reporting of values outside
```

> a range that is worth reporting on. It's very versitile, once you understand the

Reviewer: ```suggestion
a range that is worth reporting on. It's very versatile, once you fully understand the
```

> cases the lack of change needs reporting.

Reviewer: The last sentence is a bit awkward.  I would actually delete it or maybe reword something like

'In other cases, alerts are required if sensor data doesn't change over a period of time.'

Reviewer: Reading below it doesn't seem you have an example that illustrates the use case in the last sentence?

> node fully.

Reviewer: ```suggestion
node.
```

<!--more-->

> The Filter node is part of the core nodes in Node-RED, meaning they're installed

Reviewer: ```suggestion
The Filter node is part of the core nodes in Node-RED, meaning it is installed
```
